### PR TITLE
[In_app_purchase]SKProduct related fixes

### PR DESCRIPTION
**Pull request by cyanglaz**
Fixes the price property in SKProduct and SKProductDiscount losing precision after receiving from Objective-c

Fixes the example app to display error as priority.

### CODE REVIEW

File: packages/in_app_purchase/lib/src/billing_client_wrappers/sku_details_wrapper.g.dart [hunk 1]
```diff
@@ -20,15 +20,46 @@ SkuDetailsWrapper _$SkuDetailsWrapperFromJson(Map json) {
       sku: json['sku'] as String,
       subscriptionPeriod: json['subscriptionPeriod'] as String,
       title: json['title'] as String,
-      type: const SkuTypeConverter().fromJson(json['type'] as String),
+      type: _$enumDecode(_$SkuTypeEnumMap, json['type']),
```
Comment (mklim) [inline packages/in_app_purchase/lib/src/billing_client_wrappers/sku_details_wrapper.g.dart]: This ended up broken again somehow.

File: packages/in_app_purchase/lib/src/store_kit_wrappers/sk_product_wrapper.dart [hunk 2]
```diff
@@ -25,7 +25,7 @@ class SkProductResponseWrapper {
   /// This method should only be used with `map` values returned by [SKRequestMaker.startProductRequest].
   /// The `map` parameter must not be null.
   factory SkProductResponseWrapper.fromJson(Map map) {
-    assert(map != null);
+    assert(map != null, 'map must not be null.');
```
Comment (mklim) [inline packages/in_app_purchase/lib/src/store_kit_wrappers/sk_product_wrapper.dart]: nit: Capitalize "M" here and below.

File: packages/in_app_purchase/lib/src/store_kit_wrappers/sk_product_wrapper.dart [hunk 3]
```diff
@@ -146,6 +146,8 @@ class SKProductDiscountWrapper {
   /// The [subscriptionPeriod] of the discount is independent of the product's [subscriptionPeriod],
   /// and their units and duration do not have to be matched.
   final SKProductSubscriptionPeriodWrapper subscriptionPeriod;
+
+  static _priceFromJson(double price) => num.parse(price.toStringAsFixed(2));
```
Comment (mklim) [inline packages/in_app_purchase/lib/src/store_kit_wrappers/sk_product_wrapper.dart]: Are we sure we always want to round the price to two places? Just want to make sure that we're not losing precision here.

File: packages/in_app_purchase/lib/src/billing_client_wrappers/sku_details_wrapper.g.dart (hunk 1 above)
Comment (cyanglaz) [reply packages/in_app_purchase/lib/src/billing_client_wrappers/sku_details_wrapper.g.dart]: Going to revert this file

File: packages/in_app_purchase/lib/src/store_kit_wrappers/sk_product_wrapper.dart (hunk 3 above)
Comment (cyanglaz) [reply packages/in_app_purchase/lib/src/store_kit_wrappers/sk_product_wrapper.dart]: Ok, I went to check the app store connect and seems like some currencies do have 3-place decimals. Will need to think about how to handle this in a different way.

**Pull request merged**